### PR TITLE
Lay out Resolver nodes and scale with Zoom

### PR DESCRIPTION
**Pull request by bkimmel**
## Summary
This PR adds a basic layout to build on for Resolver nodes that scales with the `camera` facilities provided by the Pan/Zoom selectors and state. It incorporates EUI color palette functions for shading (and creates space for arbitrage and the provision of more complex paint servers like gradients, patterns, etc. in `/defs.tsx`).

Zoomed out:
image

Zoomed in:
image

Detail level:
image

Focus:
procnodes5

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server)
- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

### CODE REVIEW

File: x-pack/plugins/endpoint/public/embeddables/resolver/store/data/selectors.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ import {
 } from '../../models/indexed_process_tree';
 
 const unit = 100;
-const distanceBetweenNodesInUnits = 1;
+const distanceBetweenNodesInUnits = 2;
```
Comment (bkimmel) [inline x-pack/plugins/endpoint/public/embeddables/resolver/store/data/selectors.ts]: I think it needs to be at least 2 to give the nodes enough room to lay out anything useful... might think about changing it (or even making it responsive to props or something at some point)

File: x-pack/plugins/endpoint/public/embeddables/resolver/view/process_event_dot.tsx [hunk 2]
```diff
@@ -39,32 +40,95 @@ export const ProcessEventDot = styled(
        */
       const projectionMatrix = useSelector(selectors.projectionMatrix);
       const [left, top] = applyMatrix3(position, projectionMatrix);
+
+      const [magFactorX] = projectionMatrix;
```
Comment (bkimmel) [inline x-pack/plugins/endpoint/public/embeddables/resolver/view/process_event_dot.tsx]: There is an assumption being made here: namely that X (the term pulled out of the matrix here) will not scale independently from Y. Spoke about it with Robert and he said that seems like a pretty safe bet - it's hard to imagine why we would want the scene to get distorted like that.

File: x-pack/plugins/endpoint/public/embeddables/resolver/view/index.tsx [hunk 3]
```diff
@@ -140,21 +141,23 @@ const Resolver = styled(
 
     return (
       <div data-test-subj="resolverEmbeddable" className={className}>
+        <SymbolDefinitions />
         <GraphControls />
         <div className="resolver-graph" onMouseDown={handleMouseDown} ref={refCallback}>
```
Comment (bkimmel) [inline x-pack/plugins/endpoint/public/embeddables/resolver/view/index.tsx]: I wanted this to have a `role="tree"` but the precommit a11y checks barked about needing to take focus - even though the ARIA role info for tree indicates that the element's _children_ should be focusable (they are) but doesn't say anything about the tree itself.

**Comment by oatkiller** [comment]
@bkimmel thoughts on making this look like the UX mocks?

**Pull request closed without merging**